### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by vijaytembugade**
- Added Authentication 
- Added login and signup functionality
- Added toast
- Added Private / Public routes

### CODE REVIEW

File: src/Components/Toast/Toast.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    <div>
+      <Toaster
+        position="bottom-right"
+        reverseOrder={false}
+        toastOptions={{
+          className: "bg-purple-200 text-black",
+          duration: 3000,
+        }}
+      />
+    </div>
```
Comment (msdeshmukh009) [inline src/Components/Toast/Toast.js]: You can use react fragments to wrap here

File: src/Features/Auth/Slice/authSlice.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+export const userLogin = createAsyncThunk(
+  "auth/login",
+  async ({ username, password }) => {
+    const { data } = await axios.post("/api/auth/login", {
+      username,
+      password,
+    });
+    return data;
+  }
```
Comment (msdeshmukh009) [inline src/Features/Auth/Slice/authSlice.js:24]: You can use `try{..} catch{..}` here and return the error using `rejectWithValue()` method.

**Review by msdeshmukh009 — APPROVED** [review]
The login, signup & logout functionality is working as expected. Nice work !!

File: src/App.js [hunk 3]
```diff
@@ -1,21 +1,38 @@
 import { Route, Routes } from "react-router-dom";
 import "./App.css";
 import Navbar from "./Components/Headers/Navbar";
+import Login from "./Features/Auth/Components/Login";
+import Signup from "./Features/Auth/Components/Signup";
+import Profile from "./Features/Profile/Components/Profile";
+import NewPost from "./Features/Posts/Components/NewPost";
```
Comment (t007rushi) [inline src/App.js:7]: Barrel export

File: src/App.js [hunk 4]
```diff
@@ -1,21 +1,38 @@
...
-      <div className="px-8 md:px-16 py-8 md:py-16 mt-16">
+      <div className="px-8 md:px-16 py-8 md:py-16 mt-16 ">
         <Routes>
           <Route path="/" element={<Homepage />} />
-          <Route path="/new-post" element={<NewPost />} />
-          <Route path="/userprofile" element={<Profile />} />
+          <Route
+            path="/new-post"
+            element={
+              <PrivateRoutes>
```
Comment (t007rushi) [inline src/App.js:24]: Rename to ```RequireAuth``` as per convention

File: src/Features/Auth/Components/Login.js [hunk 5]
```diff
@@ -1,6 +1,30 @@
...
 const Login = () => {
+  const authState = useSelector((store) => store.auth);
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { status, isLoggedIn } = authState;
+  const dispatch = useDispatch();
+  const from = location.state?.from || "/";
+
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
```
Comment (t007rushi) [inline src/Features/Auth/Components/Login.js:16]: Use single useState with both as key in single obj

**Review by t007rushi — APPROVED** [review]
Great start with redux
 keep it up vijay

**Pull request merged**